### PR TITLE
Skip TestAutograd.test_leaky_relu_inplace_with_neg_slope on Rocm

### LINKED ISSUE #35689: `TestAutograd.test_leaky_relu_inplace_with_neg_slope` fails intermittently on Rocm
From https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py3.6-clang7-rocmdeb-ubuntu16.04-test2/24238/console :
```
test_leaky_relu_inplace_with_neg_slope (__main__.TestAutograd) ... Memory access fault by GPU node-4 (Agent handle: 0x2091860) on address 0x7f9bac673000. Reason: Page not present or supervisor privilege.
Traceback (most recent call last):
  File "test/run_test.py", line 682, in <module>
    main()
  File "test/run_test.py", line 675, in main
    raise RuntimeError(message)
RuntimeError: test_autograd failed! Received signal: SIGIOT
```

cc @ezyang @gchanan @zou3519

### PR DESCRIPTION
**Pull request by malfet**

### CODE REVIEW

**Comment by xw285cornell** [comment]
@iotamudelta I have a few other tests that fail with this, test_allgather_ops, test_autocast_banned. This is tracked in FBA-180

**Comment by malfet** [comment]
@xw285cornell let me amend the PR to disable those as well

**Comment by malfet** [comment]
@xw285cornell `test_autocast_banned` is already marked as disabled on Rocm:
```
    @skipIfRocm
    def test_autocast_banned(self):

```

**Comment by kostmo** [comment]
Is this working properly?  ROCm error message:
```
21:54:28 ERROR: test_allgather_ops (__main__.ProcessGroupNCCLTest)
21:54:28 ----------------------------------------------------------------------
21:54:28 Traceback (most recent call last):
21:54:28   File "/var/lib/jenkins/.local/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 103, in wrapper
21:54:28     sys.exit(TEST_SKIPS['skipIfRocm'].exit_code)
21:54:28 SystemExit: 78
```

File: test/distributed/test_c10d.py [hunk 1]
```diff
@@ -1683,6 +1683,8 @@ def reduce(xs, rootRank, rootTensor):
                 torch.tensor([float(self.num_gpus * (self.num_gpus + 1) / 2)]),
                 tensors[rt])
 
+    # See https://github.com/pytorch/pytorch/issues/35689
+    @skip_if_rocm
```
Comment (ezyang) [inline test/distributed/test_c10d.py]: this looks spelled wrong??

Comment (malfet) [reply test/distributed/test_c10d.py:1687]: Why? (although I admit its a bit strange to have `skipIfRocm` in `common.py` and `skip_if_rock` in `common_distributed.py`) https://github.com/pytorch/pytorch/blob/2b068d10b005c1c5e6e4a33e899adae2d964a4ee/torch/testing/_internal/common_distributed.py#L96

**Comment by gchanan** [comment]
let's do https://github.com/pytorch/pytorch/pull/35816 instead for the neg_slope test.

**Pull request closed without merging**